symfony/plugins/orangehrmOpenidAuthenticationPlugin/lib/authproviders/GoogleAuthProvider.php

```diff
@@ -23,43 +23,49 @@
  * Description of GoogleAuthProvider
  */
 class GoogleAuthProvider extends AbstractAuthProvider {
-
     const PROFILE_SCOPE = 'profile';
     const EMAIL_SCOPE = 'email';
+    protected $loginService = null;
+    protected $option = array();
 
-    public function validateUser($provider, $authProvider = null) {
-
+    /**
+     * @param OpenidProvider $provider
+     * @param AuthProviderExtraDetails $authProvider
+     * @return array
+     */
+    public function validateUser($provider, $authProvider = null)
+    {
         $gClient = new Google_Client();
         $gClient->setApplicationName('Login to Google +');
         $gClient->setClientId($authProvider->getClientId());
         $gClient->setClientSecret($authProvider->getClientSecret());
         $gClient->setRedirectUri($provider->getProviderUrl());
         $gClient->setDeveloperKey($authProvider->getDeveloperKey());
-        $gClient->addScope(array(self::EMAIL_SCOPE,self::PROFILE_SCOPE));
-
-        $gService = new Google_Service_Plus($gClient);
+        $gClient->addScope(array(self::EMAIL_SCOPE, self::PROFILE_SCOPE));
+        $requestParameters = $this->getOption();
+        $requestCode = $requestParameters['code'];
 
-        if (isset($_GET['code'])) {
-            $gClient->authenticate($_GET['code']);
+        if (isset($requestCode)) {
+            $gClient->fetchAccessTokenWithAuthCode($requestCode);
         }
-
         if ($gClient->getAccessToken()) {
-            $user = $gService->people->get("me");
-            $emails = $user->getEmails();
-            foreach ($emails as $email) {
-                if ($email->getType() == 'account') {
-                    $primaryEmail = $email->getValue();
-                }
-            }
-            $username = $primaryEmail;
+            $tokenData = $gClient->verifyIdToken();
+            $username = $tokenData[self::EMAIL_SCOPE];
             $dataArray['providerid'] = $provider->getProviderId();
-            $dataArray['useridentity'] = $gClient->getAccessToken();
+            $dataArray['useridentity'] = json_encode($gClient->getAccessToken());
             $success = $this->getOpenIdService()->setOpenIdCredentials($username, $dataArray);
             if ($success) {
-                $flag = array('type' => 'true', 'message' => 'User has authentication!');
+                $this->getLoginService()->addLogin();
+                $flag = array(
+                    'type' => 'true',
+                    'message' => 'User has authentication!'
```

Comment:
This is plain English message. What happen for multi languages 
I think it should be __("User has authentication")